lua/swenv/project.lua

```diff
@@ -1,9 +1,39 @@
 local M = {}
+-- local Path = require('plenary.path')
+
+local function isdir(dir_path)
+  return vim.fn.isdirectory(dir_path) ~= 0
+end
+
+local function get_local_env(venv_path)
+  local venv_cfg = io.open(venv_path .. '/pyvenv.cfg', 'r')
+  if not venv_cfg then
+    return nil
+  end
+
+  local configText = venv_cfg:read('*a') -- *a or *all reads the whole file
+  local pattern =  "prompt%s*=%s*([%w_%p]+)" --"prompt%s*=%s*([%w_]+)"
+  local venv_name = configText:match(pattern)
+
+  local venv = {
+    name = venv_name,
+    path = venv_path,
+    source = 'local',
+  }
+  return venv
+end
 
 -- Get the name from a `.venv` file in the project root directory.
-M.read_venv_name = function(project_dir)
-  local file = io.open(project_dir .. '/.venv', 'r') -- r read mode
+M.get_project_venv_data = function(project_dir)
+  local venv_path = project_dir .. '/.venv'
+  local file = io.open(venv_path, 'r') -- r read mode
```

Comment:
maybe move this below the if-statement since it's not needed if the body is executed